apps/examples/board_specific_demo/ameba_wifi_csi_demo/example_wifi_csi.c

```diff
@@ -0,0 +1,142 @@
+/******************************************************************************
+ *
+ * Copyright(c) 2007 - 2015 Realtek Corporation. All rights reserved.
+ *
+ *
+ ******************************************************************************/
+ /******************************************************************************
+
+ * Copyright 2022 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+
+  ******************************************************************************/
+
+
+#include <stdio.h>
+#include <tinyara/wifi/rtk/wifi_structures.h>
+#include "example_wifi_csi.h"
+#include "sched.h"
+
+#define IP_ADDR_INVALID "0.0.0.0"
+static uint8_t * getIp(void);
+extern int wifi_csi_config(rtw_csi_action_parm_t *act_param);
+extern int wifi_csi_report(uint32_t buf_len, uint8_t *csi_buf, uint32_t *len, rtw_csi_header_t *csi_header);
+extern int wifi_is_running(unsigned char wlan_idx);
+sem_t wc_ready_sema;
+
+unsigned int csi_raw_data_len = 2048;
+int _wifi_csi_thread(int argc, char *argv[])
+{
+	rtw_csi_action_parm_t act_param = {0};
+	rtw_csi_header_t csi_hdr = {0};
+	unsigned int len;
+	unsigned char *csi_buf = NULL;
+	act_param.group_num = 0;
+	act_param.mode = 2;  /* currently only supported */
+	act_param.accuracy = 0;
+	act_param.trig_period = 200;  /* ms */
+	act_param.data_rate = 0xC;  /* ofdm 6 mpbs*/
+//	act_param.mac_addr = {0x00, 0xe0, 0x4c, 0x81, 0x92, 0xbb};
+
+	while (1) {
+		uint8_t *ip = getIp();
+		if (wifi_is_running(WLAN0_IDX) && ((wifi_get_join_status() == RTW_JOINSTATUS_SUCCESS) && (strncmp(ip, IP_ADDR_INVALID, sizeof(ip))))){
+			up_mdelay((unsigned long)2000); /* 2s */
+			break;
+		}
+		up_mdelay((unsigned long)2000); /* 2s */

```

Comment:
The `up_mdelay` is a busy-wait function.
We only use this at boot time because context switching is impossible at that time.
Let's use `sleep` instead of this.

Reply:
Will change to sleep.